lib-ui-core/src/main/java/com/pax/us/pay/ui/core/BaseActionHelper.java

```diff
@@ -91,6 +99,14 @@ protected void showUI(@Nullable IUIListener uiListener, @NonNull Bundle bundle)
             if (transType.length() > 0 || message.length() > 0) {
                 ((IMessageListener) uiListener).onShowMessage(transType, message, transMode);
             }
+
+
+        }
+        if (uiListener instanceof IHasPhyKeyboardListener) {
+            if ("A30".equals(Build.MODEL)){
+                //boolean hasPhyKeyboard = bundle.getBoolean(EntryExtraData.PARAM_HAS_PHYSICAL_KEYBOARD, false);
+                ((IHasPhyKeyboardListener) uiListener).onHasPhyKeyboard(true);
+            }
```

Comment:
this interface confuses me, cuz A30 is not the only terminal that has keyboard, PLEASE name method correctly.